docs/dapp/diagrams/c10l-smart-contract-tx.mmd

```diff
@@ -0,0 +1,49 @@
+sequenceDiagram title: Confidential Smart Contract Transaction Flow
+    participant Client as Client<br/>(e.g. Node.js,<br/>MetaMask)
+    participant Web3 as Oasis Web3<br/>gateway
+    participant ClientN as Oasis Node<br/>ParaTime client
+    participant Compute as Oasis Node<br/>ParaTime compute<br/>🔒
+    participant KM as Oasis Node<br/>Key manager<br/>🔒
+
+    Client->>Client: Create transaction<br/>body
+
+    opt Encrypt Transaction
+        Client->>+Web3: Get ephemeral<br/>public key R
+        Web3->>+ClientN: Get ephemeral<br/>public key R
+        ClientN->>+KM: Get ephemeral<br/>public key R
+        KM->>KM: Derive x25519 keypair R,R'<br/>from Master key ||<br/>Epoch || ParaTime ID
```

Comment:
Probably no need to specify the concatenation (as it is not just simply concatenated), just say ephemeral per-epoch per-runtime key.